### ISSUE TITLE
inverse strides are not consistent with transposition when zero-sized

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#77498]
### 🐛 Describe the bug

Typically inverse will produce a transposed tensor but when numel is zero it doesn't transpose the strides, thereby losing the memory format.

```
>>> import torch
>>> torch.inverse(torch.empty((1, 5, 5))).stride()
(25, 1, 5)
>>> torch.inverse(torch.empty((0, 5, 5))).stride()
(25, 5, 1)
```

### Versions

master

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano @ezyang @ngimel

**Comment by Lezcano** [comment]
We have a similar behaviour on virtually all linalg functions that are not implemented as structured kernels. The reason is that we have something like
```c++
if (input.numel() == 0) return at::zeros_like(input);
```
(with better indentation) at the beginning to handle this case separately, as BLAS/cuSolver/etc often expect non-empty tensors. Otherwise, we need batch of Fortran-contig matrices (column-major), and that's why the result is returned transposed.

In structured kernels, like `linalg.svd` and a few others, the strides are always generated as column-major matrices, so the line above reads
```c++
if (result.numel() == 0) return result;
```
so the invariant of always returning batches of F-contig matrices always holds.

**Comment by Lezcano** [comment]
fwiw, the logic to call the backends for many linalg functions is *very* (unnecessarily) convoluted. Trying to replicate this behaviour on primtorch without cleaning the original functions first will be particularly difficult.

When it comes to `linalg.inverse`, once https://github.com/pytorch/pytorch/pull/74046/ is merged (currently blocked by https://github.com/pytorch/pytorch/pull/72935#issuecomment-1121473646), I could add another PR implementing `linalg.inverse`  as `linalg.solve(A, torch.eye(n))`, so writing it in primtorch should be then easy.
